e2e/test/scenarios/actions/reproductions/32974-linked-parameter-via-url-does-not-work.cy.spec.js

```diff
@@ -0,0 +1,170 @@
+
+const TEST_TABLE = "PRODUCTS";
+
+const ID_ACTION_PARAMETER = createMockActionParameter({
+  id: "86981cc2-2589-44b5-b559-2c8bbf5bb36a",
+  name: "ID",
+  slug: "id",
+  type: "number/=",
+  target: ["variable", ["template-tag", "id"]],
+});
+
+const ID_DASHBOARD_PARAMETER = createMockParameter({
+  name: "ID",
+  slug: "id",
+  id: "9da7bdd3",
+  type: "id",
+  sectionId: "id",
+});
+
+const DASHBOARD_DETAILS = {
+  name: "action dashboard",
+  parameters: [ID_DASHBOARD_PARAMETER],
+};
+
+const MODEL_DETAILS = {
+  name: "Products model",
+  query: { "source-table": PRODUCTS_ID },
+  database: SAMPLE_DB_ID,
+  dataset: true,
+};
+
+const QUERY_ACTION = {
+  name: "Query action",
+  type: "query",
+  parameters: [ID_ACTION_PARAMETER],
+  database_id: SAMPLE_DB_ID,
+  dataset_query: {
+    type: "native",
+    native: {
+      query: `UPDATE ${TEST_TABLE} SET PRICE = 999 WHERE ID = {{ ${ID_ACTION_PARAMETER.slug} }}`,
+      "template-tags": {
+        [ID_ACTION_PARAMETER.slug]: {
+          id: ID_ACTION_PARAMETER.id,
+          "display-name": ID_ACTION_PARAMETER.name,
+          name: ID_ACTION_PARAMETER.slug,
+          type: "text",
+        },
+      },
+    },
+    database: SAMPLE_DB_ID,
+  },
+  visualization_settings: {
+    fields: {
+      [ID_ACTION_PARAMETER.id]: {
+        id: ID_ACTION_PARAMETER.id,
+        required: true,
+        fieldType: "number",
+        inputType: "number",
+      },
+    },
+  },
+};
+
+describe("Issue 32974", { tags: ["@external", "@actions"] }, () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+    cy.createQuestion(MODEL_DETAILS, {
+      wrapId: true,
+      idAlias: "modelId",
+    });
+    setActionsEnabledForDB(SAMPLE_DB_ID, true);
+  });
+
+  it("can submit query action linked with dashboard parameters (metabase#32974)", () => {
+    cy.get("@modelId").then(modelId => {
+      createAction({ ...QUERY_ACTION, model_id: modelId }).then(
+        ({ body: { id: actionId } }) => {
+          cy.wrap(actionId).as("actionId");
+        },
+      );
+    });
+
+    setupDashboard();
+
+    cy.get("@dashboardId").then(dashboardId => {
+      cy.visit({ url: `/dashboard/${dashboardId}`, qs: { id: 1 } });
+    });
+
+    cy.log("Execute action");
+    cy.button(QUERY_ACTION.name).click();
+    modal().button(QUERY_ACTION.name).click();
+
+    cy.log("Assertions");
+    getDashboardCard().findByText("999").should("exist");
+    modal().should("not.exist");
+    undoToast().findByText("Query action ran successfully").should("exist");
+  });
+});
+
+function setupDashboard() {
+  cy.createDashboard(DASHBOARD_DETAILS).then(
+    ({ body: { id: dashboardId } }) => {
+      cy.wrap(dashboardId).as("dashboardId");
+    },
+  );
+
+  cy.then(function () {
+    updateDashboardCards({
+      dashboard_id: this.dashboardId,
+      cards: [
+        {
+          card_id: this.modelId,
+          // Map dashboard parameter to PRODUCTS.ID
+          parameter_mappings: [
```

Comment:
This is where we map a dashboard parameter to a dashcard field.
![image](https://github.com/metabase/metabase/assets/1937582/f54badf3-b252-4dbf-944d-5649ae7c969f)